frontend/choose-lang-content.php

```diff
@@ -68,9 +68,24 @@ protected function get_language_from_content() {
 		else {
 			foreach ( $this->model->get_translated_taxonomies() as $taxonomy ) {
 				$tax_object = get_taxonomy( $taxonomy );
-				if ( ! empty( $tax_object ) && $var = get_query_var( $tax_object->query_var ) ) {
-					$lang = $this->model->term->get_language( $var, $taxonomy );
+
+				if ( empty( $tax_object ) ) {
+					continue;
+				}
+
+				$var = get_query_var( $tax_object->query_var );
+
+				if ( ! is_string( $var ) || empty( $var ) ) {
+					continue;
 				}
+
+				$term = get_term_by( 'slug', $var, $taxonomy );
+
+				if ( ! $term instanceof WP_Term ) {
+					continue;
+				}
+
+				$lang = $this->model->term->get_language( $term->term_id );
 			}
 		}
```

Comment:
I don't think this reproduces exactly the previous behavior (but I don't know if it's a bad or good thing 😅).

For posts, the previous code was retrieving the lang only from single pages (`is_single() || is_page()...`), the new one retrieves it even from "archives" (as you know, the queried object is also defined in these cases). However, this feels like an improvement.

Also, there is a part I don't understand:
```php
( $var = get_queried_object_id() ) || ( $var = get_query_var( 'p' ) ) || ( $var = get_query_var( 'page_id' ) ) || ( $var = get_query_var( 'attachment_id' ) )
```
I wonder why `get_queried_object_id()` would fail while the others wouldn't 🤔
So either we miss something, either this was not necessary (I believe this latter one is the most probable).

Maybe this should go in a separate small PR, at least to give us some time to test it.

Reply:
Agreed for a separate PR.